dbaccess/block.go

```diff
@@ -2,156 +2,144 @@ package dbaccess
-	query = preloadFields(query, preloadedFields)
-
 	block := &dbmodels.Block{}
-	dbResult := query.First(block)
-
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.IsDBRecordNotFoundError(dbErrors) {
+	query := db.Model(block).
+		Where("block.block_hash = ?", blockHash)
+	query = preloadFields(query, preloadedFields)
+	err = query.First()
+	if err == pg.ErrNoRows {
 		return nil, nil
 	}
-	if httpserverutils.HasDBError(dbErrors) {
-		return nil, httpserverutils.NewErrorFromDBErrors("some errors were encountered when loading block from database:",
-			dbResult.GetErrors())
+	if err != nil {
+		return nil, err
 	}
 
 	return block, nil
 }
 
 // BlocksByHashes retreives a list of blocks with the corresponding `hashes`
-func BlocksByHashes(ctx Context, hashes []string, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Block, error) {
-	db, err := ctx.db()
+func BlocksByHashes(ctx database.Context, hashes []string, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Block, error) {
+	if len(hashes) == 0 {
+		return nil, nil
+	}
+
+	db, err := ctx.DB()
 	if err != nil {
 		return nil, err
 	}
 
-	query := db.Where("block_hash in (?)", hashes)
-	query = preloadFields(query, preloadedFields)
-
 	blocks := []*dbmodels.Block{}
-	dbResult := query.Find(&blocks)
-
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.HasDBError(dbErrors) {
-		return nil, httpserverutils.NewErrorFromDBErrors("some errors were encountered when loading blocks from database:",
-			dbResult.GetErrors())
+	query := db.Model(&blocks).Where("block.block_hash in (?)", pg.In(hashes))
+	query = preloadFields(query, preloadedFields)
+	err = query.Select()
+	if err != nil {
+		return nil, err
 	}
 
 	return blocks, nil
 }
 
 // Blocks retrieves from the database up to `limit` blocks in the requested `order`, skipping the first `skip` blocks
 // If preloadedFields was provided - preloads the requested fields
-func Blocks(ctx Context, order Order, skip uint64, limit uint64, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Block, error) {
-	db, err := ctx.db()
+func Blocks(ctx database.Context, order Order, skip uint64, limit uint64, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Block, error) {
+	db, err := ctx.DB()
 	if err != nil {
 		return nil, err
 	}
 
-	query := db.
-		Offset(skip).
-		Limit(limit)
+	var blocks []*dbmodels.Block
+	query := db.Model(&blocks).
+		Offset(int(skip)).
+		Limit(int(limit))
 
 	if order != OrderUnknown {
-		query = query.Order(fmt.Sprintf("`id` %s", order))
+		query = query.Order(fmt.Sprintf("id %s", order))
 	}
 
 	query = preloadFields(query, preloadedFields)
 
-	blocks := []*dbmodels.Block{}
-	dbResult := query.Find(&blocks)
-
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.HasDBError(dbErrors) {
-		return nil, httpserverutils.NewErrorFromDBErrors("some errors were encountered when loading blocks from the database:",
-			dbResult.GetErrors())
+	err = query.Select()
+	if err != nil {
```

Comment:
I think you need to make sure err is not pg.ErrNoRows here, and `return nil, nil` in these cases. Here and anywhere else relevant.

Reply:
Done